site/learn/expert-advice/authentication/webauthn-explained.md

```diff
@@ -0,0 +1,478 @@
+---
+layout: advice
+title: WebAuthn explained
+description: An in-depth look at how WebAuthn can help you add biometric authentication to your website.
+author: Dan Moore
+image: advice/vendor-lockin/expert-advice-avoiding-authentication-system-lock-in-header-image.png
+category: Authentication
+date: 2022-09-28
+dateModified: 2022-09-28
+---
+
+Webauthn is a new way to authenticate your users using biometric, secure authentication methods. WebAuthn requires something called an authenticator. What is an authenticator? An authenticator is a hardware or software device that is used to authenticate a user, via biometric or other methods, and that follows certain standards.
+
+That's a bit tautological, isn't it? Well, yes, but read on for more details. Also, clowns.
+
+## WebAuthn Overview
+
+WebAuthn has flows of data and interaction, termed "ceremonies". These extend a network protocol to include things in the real world, such as physical hardware or human beings. There are three main ones:
+
+1. Registration, where the user associates an authenticator with their account
+1. Authentication, where the user logs in, using WebAuthn
+1. Attestation, where the authenticator is tied back to a known source, such as an operating system vendor
+
+What does an authentication ceremony look like? At a high level, the flow of interaction is:
+
+1. The user tries to login to a website
+1. The website prompts the user to authenticate
+1. The user authenticates against their authenticator, which can use biometric means like Touch ID or any other method
+1. The authenticator signs and passes a response back to the website through the browser
+1. The website verifies the authenticator response
+1. The user is logged in
+
+In general terms, there are two types of authenticators:
+
+1. Hardware authenticators, such as Yubikeys
+1. Software authenticators, built into the operating system, such as Apple's Face ID or Touch ID
```

Comment:
Authenticators built into the device like Touch ID on a MacBook are still generally considered hardware authenticators. Apple uses its dedicated Secure Enclave:tm: to store and use the keys. The keys never leave the dedicated subsystem (except when they sync to another device via iCloud 🤷 )

A software authenticator is something that is implemented _entirely_ in software and most likely used for testing. One example is the [Virtual Authenticators Tab](https://chrome.google.com/webstore/detail/virtual-authenticators-ta/gafbpmlmeiikmhkhiapjlfjgdioafmja)
 extension from Google.

As far as I'm aware, it's generally not recommended to use a software authenticator for a production use case. The ones that I've seen are more about debugging.

You mention it later in the article, but I think platform vs cross-platform is a better distinction between two types of authenticators.